`ddev get drud/ddev-platformsh` fails with panic

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Run a Diagnostic and Paste Link Here

_No response_

### Expected Behavior

`ddev get drud/ddev-platformsh` shouldn't panic

### Actual Behavior

Panic.

### Steps To Reproduce

```
$ ddev get drud/ddev-platformsh
Downloading https://api.github.com/repos/drud/ddev-platformsh/tarball/v0.6.4
v0.6.4_4272473176.tar.gz 99.83 KiB / 99.83 KiB [====================================] 100.00% 0s
panic: interface conversion: interface {} is bool, not string

goroutine 1 [running]:
github.com/drud/ddev/pkg/util.YamlToDict({0x102bf3680?, 0x140001f7560?})
	/Users/rfay/workspace/ddev/pkg/util/yamltools.go:33 +0x510
github.com/drud/ddev/pkg/util.YamlToDict({0x102bf3680?, 0x140001f74d0?})
	/Users/rfay/workspace/ddev/pkg/util/yamltools.go:38 +0x39c
github.com/drud/ddev/pkg/util.YamlToDict({0x102bf5420?, 0x140001f7260?})
	/Users/rfay/workspace/ddev/pkg/util/yamltools.go:58 +0x6d4
github.com/drud/ddev/pkg/util.YamlToDict({0x102bf5420?, 0x140001f71d0?})
	/Users/rfay/workspace/ddev/pkg/util/yamltools.go:60 +0x61c
github.com/drud/ddev/cmd/ddev/cmd.glob..func29(0x103035a00?, {0x140000ac900?, 0x1, 0x1})
	/Users/rfay/workspace/ddev/cmd/ddev/cmd/get.go:190 +0xe48
github.com/spf13/cobra.(*Command).execute(0x103035a00, {0x140000ac8d0, 0x1, 0x1})
	/Users/rfay/workspace/ddev/vendor/github.com/spf13/cobra/command.go:860 +0x4ac
github.com/spf13/cobra.(*Command).ExecuteC(0x103039100)
	/Users/rfay/workspace/ddev/vendor/github.com/spf13/cobra/command.go:974 +0x354
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/rfay/workspace/ddev/vendor/github.com/spf13/cobra/command.go:902
github.com/drud/ddev/cmd/ddev/cmd.Execute()
	/Users/rfay/workspace/ddev/cmd/ddev/cmd/root.go:143 +0x40
main.main()
	/Users/rfay/workspace/ddev/cmd/ddev/main.go:20 +0xac
```

### Anything else?

This happens when the following is in .platform.app.yaml:

```yaml
    nodejs:
      # Install node version manager
      # See https://www.npmjs.com/package/n
      n: "*"
```

--- a/cmd/ddev/cmd/config_test.go
+++ b/cmd/ddev/cmd/config_test.go
@@ -20,7 +20,7 @@ import (
 	"github.com/drud/ddev/pkg/exec"
 	"github.com/drud/ddev/pkg/testcommon"
 	asrt "github.com/stretchr/testify/assert"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 )
 
 // TestCmdConfigHooks tests that pre-config and post-config hooks run

--- a/cmd/ddev/cmd/get.go
+++ b/cmd/ddev/cmd/get.go
@@ -19,7 +19,7 @@ import (
 	"github.com/otiai10/copy"
 	"github.com/spf13/cobra"
 	"golang.org/x/oauth2"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 	"os"
 	"path/filepath"
 	"sort"

--- a/go.mod
+++ b/go.mod
@@ -31,7 +31,7 @@ require (
 	golang.org/x/term v0.3.0
 	golang.org/x/text v0.5.0
 	gopkg.in/segmentio/analytics-go.v3 v3.1.0
-	gopkg.in/yaml.v2 v2.4.0
+	gopkg.in/yaml.v3 v3.0.1
 )
 
 require (
@@ -89,7 +89,7 @@ require (
 	google.golang.org/protobuf v1.28.0 // indirect
 	gopkg.in/cheggaaa/pb.v1 v1.0.26 // indirect
 	gopkg.in/ini.v1 v1.66.2 // indirect
-	gopkg.in/yaml.v3 v3.0.1 // indirect
+	gopkg.in/yaml.v2 v2.4.0 // indirect
 )
 
 go 1.19

--- a/pkg/ddevapp/compose_yaml.go
+++ b/pkg/ddevapp/compose_yaml.go
@@ -96,7 +96,7 @@ func fixupComposeYaml(yamlStr string, app *DdevApp) (map[string]interface{}, err
 			for k, volume := range volumes {
 				// With docker-compose v1, the volume might not be a map, it might be
 				// old-style "/Users/rfay/workspace/d9/.ddev:/mnt/ddev_config:ro"
-				if volumeMap, ok := volume.(map[interface{}]interface{}); ok {
+				if volumeMap, ok := volume.(map[string]interface{}); ok {
 					if volumeMap["source"] != nil {
 						if volumeMap["source"].(string) == app.AppRoot {
 							volumeMap["source"] = "../"
@@ -111,7 +111,7 @@ func fixupComposeYaml(yamlStr string, app *DdevApp) (map[string]interface{}, err
 			}
 		}
 		// Make sure all services have our networks stanza
-		serviceMap["networks"] = map[interface{}]interface{}{
+		serviceMap["networks"] = map[string]interface{}{
 			"ddev_default": nil,
 			"default":      nil,
 		}

--- a/pkg/ddevapp/compose_yaml.go
+++ b/pkg/ddevapp/compose_yaml.go
@@ -84,11 +84,11 @@ func fixupComposeYaml(yamlStr string, app *DdevApp) (map[string]interface{}, err
 	// so it's in Docker Desktop 4.1.0.
 	// https://github.com/docker/compose/issues/8503#issuecomment-930969241
 
-	for _, service := range tempMap["services"].(map[interface{}]interface{}) {
+	for _, service := range tempMap["services"].(map[string]interface{}) {
 		if service == nil {
 			continue
 		}
-		serviceMap := service.(map[interface{}]interface{})
+		serviceMap := service.(map[string]interface{})
 
 		// Find any services that have bind-mount to app.AppRoot and make them relative
 		if serviceMap["volumes"] != nil {

--- a/pkg/ddevapp/compose_yaml.go
+++ b/pkg/ddevapp/compose_yaml.go
@@ -3,7 +3,7 @@ package ddevapp
 import (
 	"github.com/drud/ddev/pkg/dockerutil"
 	"github.com/drud/ddev/pkg/util"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 	"os"
 	"strings"
 	//compose_cli "github.com/compose-spec/compose-go/cli"

--- a/pkg/ddevapp/config.go
+++ b/pkg/ddevapp/config.go
@@ -25,7 +25,7 @@ import (
 	"github.com/drud/ddev/pkg/output"
 	"github.com/drud/ddev/pkg/util"
 	log "github.com/sirupsen/logrus"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 )
 
 // Regexp pattern to determine if a hostname is valid per RFC 1123.

--- a/pkg/ddevapp/ddevapp.go
+++ b/pkg/ddevapp/ddevapp.go
@@ -1971,9 +1971,9 @@ func (app *DdevApp) Pause() error {
 // WaitForServices waits for all the services in docker-compose to come up
 func (app *DdevApp) WaitForServices() error {
 	var requiredContainers []string
-	if services, ok := app.ComposeYaml["services"].(map[interface{}]interface{}); ok {
+	if services, ok := app.ComposeYaml["services"].(map[string]interface{}); ok {
 		for k := range services {
-			requiredContainers = append(requiredContainers, k.(string))
+			requiredContainers = append(requiredContainers, k)
 		}
 	} else {
 		util.Failed("unable to get required startup services to wait for")

--- a/pkg/ddevapp/ddevapp.go
+++ b/pkg/ddevapp/ddevapp.go
@@ -1371,8 +1371,8 @@ func (app *DdevApp) FindAllImages() ([]string, error) {
 		return images, nil
 	}
 	if y, ok := app.ComposeYaml["services"]; ok {
-		for _, v := range y.(map[interface{}]interface{}) {
-			if i, ok := v.(map[interface{}]interface{})["image"]; ok {
+		for _, v := range y.(map[string]interface{}) {
+			if i, ok := v.(map[string]interface{})["image"]; ok {
 				if strings.HasSuffix(i.(string), "-built") {
 					i = strings.TrimSuffix(i.(string), "-built")
 					if strings.HasSuffix(i.(string), "-"+app.Name) {
@@ -2302,8 +2302,8 @@ func deleteServiceVolumes(app *DdevApp) {
 	var err error
 	y := app.ComposeYaml
 	if s, ok := y["volumes"]; ok {
-		for _, v := range s.(map[interface{}]interface{}) {
-			vol := v.(map[interface{}]interface{})
+		for _, v := range s.(map[string]interface{}) {
+			vol := v.(map[string]interface{})
 			if vol["external"] == true {
 				continue
 			}
@@ -2635,12 +2635,12 @@ func (app *DdevApp) CheckAddonIncompatibilities() error {
 		return nil
 	}
 	// Look for missing "networks" stanza and request it.
-	for s, v := range app.ComposeYaml["services"].(map[interface{}]interface{}) {
-		x := v.(map[interface{}]interface{})
+	for s, v := range app.ComposeYaml["services"].(map[string]interface{}) {
+		x := v.(map[string]interface{})
 		errMsg := fmt.Errorf("service '%s' does not have the 'networks: [default, ddev_default]' stanza, required since v1.19, please add it, see %s", s, "https://ddev.readthedocs.io/en/latest/users/extend/custom-compose-files/#docker-composeyaml-examples")
-		var nets map[interface{}]interface{}
+		var nets map[string]interface{}
 		ok := false
-		if nets, ok = x["networks"].(map[interface{}]interface{}); !ok {
+		if nets, ok = x["networks"].(map[string]interface{}); !ok {
 			return errMsg
 		}
 		// Make sure both "default" and "ddev" networks are in there.

--- a/pkg/ddevapp/ddevapp.go
+++ b/pkg/ddevapp/ddevapp.go
@@ -1396,9 +1396,9 @@ func (app *DdevApp) FindMaxTimeout() int {
 		return defaultContainerTimeout
 	}
 	if y, ok := app.ComposeYaml["services"]; ok {
-		for _, v := range y.(map[interface{}]interface{}) {
-			if i, ok := v.(map[interface{}]interface{})["healthcheck"]; ok {
-				if timeout, ok := i.(map[interface{}]interface{})["timeout"]; ok {
+		for _, v := range y.(map[string]interface{}) {
+			if i, ok := v.(map[string]interface{})["healthcheck"]; ok {
+				if timeout, ok := i.(map[string]interface{})["timeout"]; ok {
 					duration, err := time.ParseDuration(timeout.(string))
 					if err != nil {
 						continue
@@ -2636,9 +2636,9 @@ func (app *DdevApp) CheckAddonIncompatibilities() error {
 	}
 	// Look for missing "networks" stanza and request it.
 	for s, v := range app.ComposeYaml["services"].(map[string]interface{}) {
-		x := v.(map[string]interface{})
 		errMsg := fmt.Errorf("service '%s' does not have the 'networks: [default, ddev_default]' stanza, required since v1.19, please add it, see %s", s, "https://ddev.readthedocs.io/en/latest/users/extend/custom-compose-files/#docker-composeyaml-examples")
 		var nets map[string]interface{}
+		x := v.(map[string]interface{})
 		ok := false
 		if nets, ok = x["networks"].(map[string]interface{}); !ok {
 			return errMsg

--- a/pkg/ddevapp/ddevapp.go
+++ b/pkg/ddevapp/ddevapp.go
@@ -10,7 +10,7 @@ import (
 	"github.com/mattn/go-isatty"
 	"github.com/otiai10/copy"
 	"golang.org/x/term"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 	"os"
 	"path/filepath"
 	"runtime"

--- a/pkg/ddevapp/ddevapp_test.go
+++ b/pkg/ddevapp/ddevapp_test.go
@@ -3106,9 +3106,9 @@ func TestMultipleComposeFiles(t *testing.T) {
 	require.True(t, len(app.ComposeYaml) > 0)
 
 	// Verify that the env var DUMMY_BASE got set by docker-compose.override.yaml
-	if services, ok := app.ComposeYaml["services"].(map[interface{}]interface{}); ok {
-		if w, ok := services["web"].(map[interface{}]interface{}); ok {
-			if env, ok := w["environment"].(map[interface{}]interface{}); ok {
+	if services, ok := app.ComposeYaml["services"].(map[string]interface{}); ok {
+		if w, ok := services["web"].(map[string]interface{}); ok {
+			if env, ok := w["environment"].(map[string]interface{}); ok {
 				// The docker-compose.override should have won with the value of DUMMY_BASE
 				assert.Equal("override", env["DUMMY_BASE"])
 				// But each of the DUMMY_COMPOSE_ONE/TWO/OVERRIDE which are unique

--- a/pkg/ddevapp/provider.go
+++ b/pkg/ddevapp/provider.go
@@ -12,7 +12,7 @@ import (
 	"github.com/drud/ddev/pkg/fileutil"
 	"github.com/drud/ddev/pkg/util"
 
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 )
 
 // ProviderCommand defines the shell command to be run for one of the commands (db pull, etc.)

--- a/pkg/ddevapp/traefik.go
+++ b/pkg/ddevapp/traefik.go
@@ -31,9 +31,9 @@ func detectAppRouting(app *DdevApp) ([]TraefikRouting, error) {
 	// app.ComposeYaml["services"];
 	var table []TraefikRouting
 	if services, ok := app.ComposeYaml["services"]; ok {
-		for serviceName, s := range services.(map[interface{}]interface{}) {
-			service := s.(map[interface{}]interface{})
-			if env, ok := service["environment"].(map[interface{}]interface{}); ok {
+		for serviceName, s := range services.(map[string]interface{}) {
+			service := s.(map[string]interface{})
+			if env, ok := service["environment"].(map[string]interface{}); ok {
 				var virtualHost string
 				var ok bool
 				if virtualHost, ok = env["VIRTUAL_HOST"].(string); ok {
@@ -42,7 +42,7 @@ func detectAppRouting(app *DdevApp) ([]TraefikRouting, error) {
 				hostnames := strings.Split(virtualHost, ",")
 				if httpExpose, ok := env["HTTP_EXPOSE"].(string); ok {
 					util.Debug("HTTP_EXPOSE=%v for %s", httpExpose, serviceName)
-					routeEntries, err := processHTTPExpose(serviceName.(string), httpExpose, false, hostnames)
+					routeEntries, err := processHTTPExpose(serviceName, httpExpose, false, hostnames)
 					if err != nil {
 						return nil, err
 					}
@@ -51,7 +51,7 @@ func detectAppRouting(app *DdevApp) ([]TraefikRouting, error) {
 
 				if httpsExpose, ok := env["HTTPS_EXPOSE"].(string); ok {
 					util.Debug("HTTPS_EXPOSE=%v for %s", httpsExpose, serviceName)
-					routeEntries, err := processHTTPExpose(serviceName.(string), httpsExpose, true, hostnames)
+					routeEntries, err := processHTTPExpose(serviceName, httpsExpose, true, hostnames)
 					if err != nil {
 						return nil, err
 					}

--- a/pkg/globalconfig/global_config.go
+++ b/pkg/globalconfig/global_config.go
@@ -6,7 +6,7 @@ import (
 	"github.com/drud/ddev/pkg/nodeps"
 	"github.com/drud/ddev/pkg/output"
 	"github.com/sirupsen/logrus"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 	"net"
 	"os"
 	"os/exec"

--- a/pkg/util/yaml_test.go
+++ b/pkg/util/yaml_test.go
@@ -20,7 +20,7 @@ func TestYamlFileToMap(t *testing.T) {
 	//runtime:
 	//  extensions:
 	//	  - redis
-	runtime, ok := m["runtime"].(map[interface{}]interface{})
+	runtime, ok := m["runtime"].(map[string]interface{})
 	require.True(t, ok)
 	extensions, ok := runtime["extensions"].([]interface{})
 	require.True(t, ok)
@@ -28,7 +28,7 @@ func TestYamlFileToMap(t *testing.T) {
 
 	//relationships:
 	//  database: 'db:mysql'
-	relationships, ok := m["relationships"].(map[interface{}]interface{})
+	relationships, ok := m["relationships"].(map[string]interface{})
 	require.True(t, ok)
 	assert.Equal("db:mysql", relationships["database"])
 
@@ -36,11 +36,11 @@ func TestYamlFileToMap(t *testing.T) {
 	//  locations:
 	//	  '/':
 	//		root: 'web'
-	web, ok := m["web"].(map[interface{}]interface{})
+	web, ok := m["web"].(map[string]interface{})
 	require.True(t, ok)
-	locations, ok := web["locations"].(map[interface{}]interface{})
+	locations, ok := web["locations"].(map[string]interface{})
 	require.True(t, ok)
-	slashloc, ok := locations["/"].(map[interface{}]interface{})
+	slashloc, ok := locations["/"].(map[string]interface{})
 	require.True(t, ok)
 	assert.Equal("web", slashloc["root"])
 

--- a/pkg/util/yamltools.go
+++ b/pkg/util/yamltools.go
@@ -30,7 +30,12 @@ func YamlToDict(topm interface{}) (map[string]interface{}, error) {
 	switch topm.(type) {
 	case map[interface{}]interface{}:
 		for yk, v := range topm.(map[interface{}]interface{}) {
-			ys := yk.(string)
+			ys, ok := yk.(string)
+			// TODO: Explain the problem, figure out if this is a golang/yaml bug
+			if !ok {
+				Warning("%v can't be converted to string")
+				continue
+			}
 			switch v.(type) {
 			case string:
 				res[ys] = v

--- a/pkg/util/yamltools.go
+++ b/pkg/util/yamltools.go
@@ -2,7 +2,7 @@ package util
 
 import (
 	"fmt"
-	"gopkg.in/yaml.v2"
+	"gopkg.in/yaml.v3"
 	"os"
 )
 

--- a/pkg/util/yamltools.go
+++ b/pkg/util/yamltools.go
@@ -28,39 +28,11 @@ func YamlToDict(topm interface{}) (map[string]interface{}, error) {
 	var err error
 
 	switch topm.(type) {
-	case map[interface{}]interface{}:
-		for yk, v := range topm.(map[interface{}]interface{}) {
-			ys, ok := yk.(string)
-			// TODO: Explain the problem, figure out if this is a golang/yaml bug
-			if !ok {
-				Warning("%v can't be converted to string")
-				continue
-			}
-			switch v.(type) {
-			case string:
-				res[ys] = v
-			case map[interface{}]interface{}:
-				res[ys], err = YamlToDict(v)
-			case map[string]interface{}:
-				res[ys], err = YamlToDict(v)
-			case []interface{}:
-				res[ys] = v
-			case interface{}:
-				res[ys] = v
-			default:
-				return nil, fmt.Errorf("YamlToDict: type %T not handled (%v)", yk, yk)
-			}
-			if err != nil {
-				return nil, err
-			}
-		}
 	case map[string]interface{}:
 		for yk, v := range topm.(map[string]interface{}) {
 			switch v.(type) {
 			case string:
 				res[yk] = v
-			case map[interface{}]interface{}:
-				res[yk], err = YamlToDict(v)
 			case map[string]interface{}:
 				res[yk], err = YamlToDict(v)
 			case interface{}:

